cmd/tomltestgen/main.go

```diff
@@ -138,61 +139,56 @@ func main() {
 	flag.Usage = usage
 	flag.Parse()
 
-	url := "https://codeload.github.com/BurntSushi/toml-test/zip/" + *ref
-	resultFile := downloadTmpFile(url)
-	defer os.Remove(resultFile)
-	log.Println("file written to", resultFile)
-
-	zipReader, err := zip.OpenReader(resultFile)
-	if err != nil {
-		panic(err)
-	}
-	defer zipReader.Close()
-
 	collection := testsCollection{
 		Ref:       *ref,
 		Timestamp: time.Now().Format(time.RFC3339),
 	}
 
-	zipFilesMap := map[string]*zip.File{}
+	dirContent, _ := filepath.Glob("tests/invalid/**/*.toml")
+	for _, f := range dirContent {
+		filename := path.Base(f)
```

Comment:
I think the duplicated test issue comes from here. The previously used regular expression includes any directory after `invalid|valid` in the name, while this code only looks at the file name, discarding intermediate directories. As a result, it has the same name for `invalid/array/no-close-1` and `invalid/string/no-close-1`.

Reply:
Gotcha.